flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorFactory.java

```diff
@@ -88,6 +89,21 @@ private static IndexGenerator createRuntimeIndexGenerator(
                 index.substring(indexPrefix.length() + dynamicIndexPatternStr.length());
 
         final boolean isDynamicIndexWithFormat = indexHelper.checkIsDynamicIndexWithFormat(index);
+        final boolean isDynamicIndexWithProcTimeFormat =
```

Comment:
This new block can be moved right after `indexPrefix` set at line 88.
Why don't we need indexSuffix in the case of `now()`?